File: java/new_Job.java


import java.io.IOException;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.text.SimpleDateFormat;
import java.util.Date;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import JavaBean.Jdbc;


@WebServlet("/new_Job")
public class new_Job extends HttpServlet {
	private static final long serialVersionUID = 1L;
       

    public new_Job() {
        super();
    }


	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		String job_name = request.getParameter("job_name");
		String job_info = request.getParameter("job_info");
		String job_city = request.getParameter("job_city");
		String job_district = request.getParameter("job_district");
		String job_require = request.getParameter("job_require");
		String salary_type = request.getParameter("salary_type");
		
		Date dNow = new Date( );
        SimpleDateFormat ft = new SimpleDateFormat ("yyyy-MM-dd");
	      
		int salary = Integer.parseInt(request.getParameter("salary"));
		
		int userId = Integer.parseInt(request.getParameter("id"));
		Jdbc jd = null;
		try {
			jd = new Jdbc();
		} catch (ClassNotFoundException e1) {
			// TODO Auto-generated catch block
			e1.printStackTrace();
		} catch (SQLException e1) {
			// TODO Auto-generated catch block
			e1.printStackTrace();
		}
		try {
			Connection conn = jd.get_Connection();
			String sql = "select chicken_job.create_job(?,?,?,?,?,?,?,?,?)";
			PreparedStatement st = conn.prepareStatement(sql);
			st.setInt(1,userId);
			st.setString(2,job_name);
			st.setString(3,job_info);
			st.setString(4,job_city);
			st.setString(5,job_district);
			st.setString(6,ft.format(dNow));
			st.setString(7,job_require);
			st.setString(8,salary_type);
			st.setInt(9,salary);
			ResultSet rs = st.executeQuery();
			response.sendRedirect("member_center?id="+ userId);
		} catch (SQLException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		
		
	}

	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		doGet(request, response);
	}

}

File: webapp/source/js/newJob.js
function doCheckJobInfo() 
{   
    // 檢查職缺名稱
    var job_name = document.getElementById("job_name").value;
    
    if(job_name==="")
    {
    alert("請輸入職缺名稱！");
      document.getElementById("job_name").focus();
      return false;
    }
    
    //檢查職缺資訊
    var job_info = document.getElementById("job_info").value;
    if (job_info === "") 
    {
      alert("請輸入職缺資訊！");
      document.getElementById("job_info").focus();
      return false;
    }

    //檢查工作城市
    var job_city = document.getElementById("job_city").value;
    if (job_city === "") 
    {
      alert("請選擇工作城市！");
      document.getElementById("job_city").focus();
      return false;
    }

    //檢查工作區域
    var job_district = document.getElementById("job_district").value;
    if (job_district === "") 
    {
      alert("請選擇工作區域！");
      document.getElementById("job_district").focus();
      return false;
    }
    
    //檢查學歷需求
    var job_require = document.getElementById("job_require").value;
    if (job_require === "") 
    {
      alert("請選擇學歷需求！");
      document.getElementById("job_require").focus();
      return false;
    }

    //檢查支薪類型
    var salary_type = document.getElementById("salary_type").value;
    if (salary_type === "") 
    {
      alert("請選擇支薪類型！");
      document.getElementById("salary_type").focus();
      return false;
    }
    
    //薪資金額
    var salary = document.getElementById("salary").value;
    var regex = /^[1-9]\d*$/; // 限制只能輸入正整數數字，第一數字不能為0
    if (salary === "") 
    {
      alert("請輸入薪資金額");
      document.getElementById("salary").focus();
      return false;
    }
    if (!regex.test(salary)) 
    {
      alert("薪資金額只能為非0開頭的正整數！");
      document.getElementById("salary").focus();
      return false;
    }
    document.getElementById('newjob').submit();
    
}